Build fails when configuring with CMake to use Makefiles

Steps to reproduce:

```
% cmake -G 'Unix Makefiles' . && make
...
[ 52%] Generating WaylandProtocols.dir/fullscreen-shell-unstable-v1-client.h
Could not open output file: No such file or directory
make[2]: *** [CMakeFiles/cogplatform-fdo.dir/build.make:74: WaylandProtocols.dir/fullscreen-shell-unstable-v1-client.h] Error 1
make[1]: *** [CMakeFiles/Makefile2:109: CMakeFiles/cogplatform-fdo.dir/all] Error 2
```

(Issue reported by @bertogg — thanks!)

--- a/cmake/FindWaylandProtocols.cmake
+++ b/cmake/FindWaylandProtocols.cmake
@@ -117,6 +117,7 @@ function(add_wayland_protocol _target _kind _protocol)
         add_custom_command(
             OUTPUT "${proto_code}"
             MAIN_DEPENDENCY "${proto_file}"
+            COMMAND "${CMAKE_COMMAND}" -E make_directory "${CMAKE_BINARY_DIR}/WaylandProtocols.dir"
             COMMAND "${WAYLAND_SCANNER}" "${WAYLAND_SCANNER_CODE_ARG}" "${proto_file}" "${proto_code}"
             VERBATIM
         )
@@ -131,6 +132,7 @@ function(add_wayland_protocol _target _kind _protocol)
             add_custom_command(
                 OUTPUT "${proto_client}"
                 MAIN_DEPENDENCY "${proto_file}"
+                COMMAND "${CMAKE_COMMAND}" -E make_directory "${CMAKE_BINARY_DIR}/WaylandProtocols.dir"
                 COMMAND "${WAYLAND_SCANNER}" client-header "${proto_file}" "${proto_client}"
                 VERBATIM
             )
@@ -144,6 +146,7 @@ function(add_wayland_protocol _target _kind _protocol)
             add_custom_command(
                 OUTPUT "${proto_server}"
                 MAIN_DEPENDENCY "${proto_file}"
+                COMMAND "${CMAKE_COMMAND}" -E make_directory "${CMAKE_BINARY_DIR}/WaylandProtocols.dir"
                 COMMAND "${WAYLAND_SCANNER}" server-header "${proto_file}" "${proto_server}"
                 VERBATIM
             )

